lib/declare_schema/model.rb

```diff
@@ -126,13 +114,35 @@ def belongs_to(name, scope = nil, **options)
           ActiveSupport::Deprecation.warn("belongs_to limit: is deprecated since it is now inferred")
         end
 
-        index_options = {}
-        index_options[:name]   = options.delete(:index) if options.has_key?(:index)
-        index_options[:unique] = options.delete(:unique) if options.has_key?(:unique)
-        index_options[:allow_equivalent] = options.delete(:allow_equivalent) if options.has_key?(:allow_equivalent)
+        # index: true means create an index on the foreign key
+        # index: false means do not create an index on the foreign key
+        # index: { ... } means create an index on the foreign key with the given options
+        index_value = options.delete(:index)
+        if index_value != false || options.has_key?(:unique) || options.has_key?(:allow_equivalent)
+          index_options = {} # truthy iff we want an index
+          case index_value
+          when String, Symbol
+            Kernel.warn("belongs_to index: 'name' is deprecated; use index: { name: 'name' } instead (in #{name})")
+            index_options[:name] = index_value.to_s
+          when false
+            raise ArgumentError, "belongs_to index: false contradicts others options #{options.inspect} (in #{name})"
```

Comment:
These raises feel like breaking changes.

Are we sure we should include them in this PR?

Reply:
You're right, this belonged only in the feature branch. And I changed it to a warning over there.